pyfr/plugins/tavg.py

```diff
@@ -118,23 +122,12 @@ def _eval_acc_exprs(self, intg):
             # Prepare the substitutions dictionary
             subs = dict(zip(pnames, psolns))
 
-            # Compute any required gradients
+            # Prepare any required gradient
             if self._gradpnames:
-                # Gradient operator and J^-T matrix
-                gradop, rcpjact = self._gradop[i], self._rcpjact[i]
-                nupts = gradop.shape[1]
-
+                gradps = grads_eles[j]
                 for pname in self._gradpnames:
-                    psoln = subs[pname]
-
-                    # Compute the transformed gradient
-                    tgradpn = gradop @ psoln
-                    tgradpn = tgradpn.reshape(self.ndims, nupts, -1)
-
-                    # Untransform this to get the physical gradient
-                    gradpn = np.einsum('ijkl,jkl->ikl', rcpjact, tgradpn)
-                    gradpn = gradpn.reshape(self.ndims, nupts, -1)
-
+                    idx = self.elementscls.privarmap[self.ndims].index(pname)
+                    gradpn = gradps[idx][..., rgn]
```

Comment:
I think we've already subsetted the gradients here?

Reply:
I had to update the implementation in `tavg`. Now it is more consistent with `integrate`, let me know if you think it is OK